hikari/impl/bot.py

```diff
@@ -23,589 +23,721 @@
 
 from __future__ import annotations
 
-__all__: typing.Final[typing.List[str]] = ["BotApp"]
+__all__: typing.List[str] = ["BotApp", "LoggerLevel"]
 
 import asyncio
-import contextlib
+import concurrent.futures
 import datetime
 import logging
 import math
 import signal
 import sys
-import time
+import types
 import typing
-import warnings
 
 from hikari import config
 from hikari import errors
 from hikari import intents as intents_
 from hikari import presences
 from hikari import traits
-from hikari import undefined
+from hikari import users
+from hikari.api import cache as cache_
+from hikari.api import chunker as chunker_
+from hikari.api import entity_factory as entity_factory_
 from hikari.api import event_dispatcher
+from hikari.api import event_factory as event_factory_
 from hikari.api import shard as gateway_shard
+from hikari.api import voice as voice_
 from hikari.events import lifetime_events
 from hikari.impl import entity_factory as entity_factory_impl
 from hikari.impl import event_factory as event_factory_impl
-from hikari.impl import rest as rest_client_impl
-from hikari.impl import shard as gateway_shard_impl
-from hikari.impl import stateful_cache as cache_impl
-from hikari.impl import stateful_event_manager
-from hikari.impl import stateful_guild_chunker as guild_chunker_impl
-from hikari.impl import stateless_cache as stateless_cache_impl
-from hikari.impl import stateless_event_manager
-from hikari.impl import stateless_guild_chunker as stateless_guild_chunker_impl
-from hikari.impl import voice
-from hikari.utilities import art
+from hikari.impl import rest as rest_impl
+from hikari.impl import shard as shard_impl
+from hikari.impl import voice as voice_impl
+from hikari.utilities import aio
 from hikari.utilities import constants
 from hikari.utilities import date
-from hikari.utilities import version_sniffer
+from hikari.utilities import event_stream
+from hikari.utilities import ux
 
 if typing.TYPE_CHECKING:
-    import concurrent.futures
-
-    from hikari import users
-    from hikari.api import cache as cache_
-    from hikari.api import chunker as guild_chunker_
-    from hikari.events import base_events
+    from hikari.api import cache
+    from hikari.api import chunker
+    from hikari.api import rest as rest_
+    from hikari.api import shard
     from hikari.impl import event_manager_base
-    from hikari.utilities import event_stream
+
+LoggerLevel = typing.Union[
```

Comment:
This feels a bit funky to me, being PascalCase but without T at the end of it to identify it as solely a type variable

Reply:
fixed.